language/vm/src/file_format.rs

```diff
@@ -1826,8 +1826,8 @@ pub struct CompiledModuleMut {
     /// Handles to fields.
     pub field_handles: Vec<FieldHandle>,
 
-    /// Friend declarations.
-    pub friend_decls: Vec<ModuleHandleIndex>,
+    /// Friend declarations, represented as a collection of handles to external friend modules.
+    pub friend_decls: Vec<ModuleHandle>,
```

Comment:
mega nit, I didn't realize this last time, but there should be no whitespace on 1828, and then there should be one after friend_decls

friend_decls does not belong to the next grouping, which is for generic instantiations

Reply:
Fixed